cg/models/fohm/reports.py

```diff
@@ -0,0 +1,36 @@
+"""FOHM report models."""
+
+from pydantic import BaseModel, Field
+
+
+class FohmComplementaryReport(BaseModel):
+    """Model for validating FOHM complementary reports."""
+
+    gisaid_accession: str = Field(str, alias="GISAID_accession")
+    internal_id: str | None = None
+    sample_number: str = Field(str, alias="provnummer")
+    selection_criteria: str = Field(str, alias="urvalskriterium")
+    region_lab: str | None = None
+
+
+class FohmPangolinReport(BaseModel):
```

Comment:
Pangolin is means nothing to us devs 😅 
```suggestion
class FohmSampleResults(BaseModel):
```

Reply:
Me neither 😅. Any suggestions @Clinical-Genomics/prodbioinfo?

Reply:
Pangolin is the tool used to assign lineages to strains https://cov-lineages.org/resources/pangolin.html

So to us in prod bioinfo it's at least clear 😛 Maybe FohmTypingReport or FohmLineageReport if you feel the need to rename it. 

For context: the output file is called `<ticket>.pangolin.csv` I believe, so in that context the old name maybe make more sense?

Reply:
I think we should keep it coupled to the software, i.e. Pangolin in the name if it is meant to validate such a report.

Reply:
What if the tool is swapped out? If there is logic tied to this specific tool it is better to abstract it away

Reply:
Maybe we can have a `LineageReportBuilder` or something and a concrete implementation of it in the form of a `PangolinReportBuilder` or whatever

Reply:
Another PR.